src/backend/distributed/planner/extended_op_node_utils.c

```diff
@@ -180,6 +184,22 @@ HasNonPartitionColumnDistinctAgg(List *targetEntryList, Node *havingQual,
 			continue;
 		}
 
+		/*
+		 * We are dealing with a more complex count distinct, it needs to be
+		 * evaluated at coordinator level.
+		 */
+		if (list_length(targetAgg->args) > 1 || list_length(targetAgg->aggdistinct) > 1)
+		{
+			return true;
+		}
+
+		firstTargetEntry = linitial_node(TargetEntry, targetAgg->args);
+		firstTargetExprNode = strip_implicit_coercions((Node *) firstTargetEntry->expr);
+		if (!IsA(firstTargetExprNode, Var))
```

Comment:
The description of this function should be updated to reflect the change.

I think it might also be worth adding a comment that we typically cannot push down expressions because we cannot easily tell whether an expression maps two distinct partition column values to the same value.

Reply:
ok